### PR TITLE
feat: expose Gemini adapter and model catalog discovery across adapters

### PR DESCRIPTION
**Pull request by phrazzld**
## Summary
This PR completes Moonbridge's multi-adapter exposure path by adding first-class Gemini support and making adapter/model discovery explicit in MCP.

## Problem
Moonbridge runtime already had multiple adapters, but MCP discoverability was biased toward the default adapter and did not provide a dedicated model catalog surface.

Key gaps:
- `check_status` only targeted the default adapter.
- Tool metadata was framed as adapter-specific rather than platform-wide.
- No direct MCP tool existed for model-range discovery.
- Gemini CLI was not integrated as an adapter.

## Design (Ousterhout-oriented)
- Keep adapter-specific behavior inside adapters, not protocol handlers.
- Extend `CLIAdapter` with a small, cohesive interface: `list_models(...)`.
- Keep MCP protocol simple and explicit:
  - `check_status(adapter?)` for targeted health checks.
  - `list_models(adapter?, provider?, refresh?)` for model discovery.
- Avoid scattered routing logic by centralizing model-catalog orchestration in `server._model_catalog`.

### 1) Adapter layer
- Added `GeminiAdapter`:
  - non-interactive command construction (`gemini -p ... --output-format text --approval-mode yolo`)
  - auth/install hints and safe env allowlist
  - default model + known models + timeout defaults
- Added `list_models(...)` to `CLIAdapter` protocol.
- Implemented `list_models(...)` across adapters:
  - `kimi`, `codex`, `gemini`: static catalog from adapter config.
  - `opencode`: dynamic catalog via `opencode models [provider] [--refresh]`.
- Registered `gemini` in `ADAPTER_REGISTRY`.

### 2) MCP surface and orchestration
- Added MCP tool: `list_models`.
- Updated `check_status` to accept optional `adapter` override.
- Updated tool schema generation:
  - adapter-neutral descriptions
  - dynamic adapter enums for `spawn_agent`, `spawn_agents_parallel`, `list_models`, and `check_status`
- Added `_model_catalog(...)` in server with:
  - install checks
  - provider guardrails (`provider` only for OpenCode)
  - stable success/error payloads
  - de-duplication of discovered model names

### 3) Docs/config
- README updated for:
  - Gemini CLI install/auth
  - new `list_models` tool
  - `check_status.adapter`
  - expanded adapter/model/timeout env vars
- CLAUDE.md updated for architecture/tooling/env accuracy.
- `.env.example` expanded with adapter + model + timeout defaults.

## Tests
Added and updated tests for:
- Gemini adapter command/config/install behavior.
- OpenCode dynamic model discovery (success + failure).
- Tool schema now exposing all adapters (including Gemini) across all relevant tools.
- `check_status` adapter override.
- `list_models` success/error/provider-guard cases.

## Verification
All quality gates pass:
- `uv run ruff check src tests`
- `uv run mypy src`
- `uv run pytest -q` (239 passed)

## Commits
1. `feat(adapters): add gemini adapter and model catalog interface`
2. `feat(mcp): add model catalog tool and adapter-scoped status checks`
3. `docs: document multi-adapter routing and model discovery`

## Summary by CodeRabbit

* **New Features**
  * Added Gemini and OpenCode adapters and a new "list_models" tool; check_status can target a specific adapter.

* **Configuration**
  * Added per-adapter and global env/config options for model defaults, per-adapter timeouts, parallel agents, max output length, allowed dirs, and log level.

* **Documentation**
  * Updated README/docs with Gemini/OpenCode setup, auth, defaults, tooling, and troubleshooting.

* **Tests**
  * Expanded tests for new adapters, dynamic model listing, provider filters, and adapter override flows.

* **Chores**
  * CI workflow updated to newer action versions.

---

### Before
- Working PR with 239 passing tests, but three adapters (kimi/codex/gemini) had identical 12-line `list_models` bodies (copy-paste smell)
- Provider-filter guard in `_model_catalog` hardcoded to string `"opencode"` — server knew adapter internals
- `REASONING_EFFORT_PARAM_SHORT` was byte-identical to `REASONING_EFFORT_PARAM` (dead distinction)
- Missing edge-case tests for ANSI stripping, provider flag injection, deduplication, empty output, exception handling

### After
- Extracted `static_model_catalog()` helper in `base.py` — three adapters delegate in one line
- Added `supports_provider_filter` flag to `AdapterConfig`; server uses capability, not name
- Collapsed redundant param constant into an alias
- 14 new tests (253 total); all edge cases covered
- Two architectural issues filed for future work: #103 (protocol shape), #104 (catalog staleness)

### CODE REVIEW

File: src/moonbridge/tool_handlers.py [hunk 1]
```diff
@@ -204,8 +205,16 @@ async def handle_tool(
...
+                provider_raw = arguments.get("provider")
+                provider = provider_raw.strip() if isinstance(provider_raw, str) else None
+                provider = provider or None
+                refresh = bool(arguments.get("refresh", False))
+                return json_text(deps.model_catalog(cwd, adapter, provider, refresh))
+
             if name == "check_status":
-                adapter = deps.get_adapter(None)
+                adapter = deps.get_adapter(arguments.get("adapter"))
                 return json_text(deps.status_check(cwd, adapter))
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply src/moonbridge/tool_handlers.py]: Fixed in 4497f2e — `check_status` now wrapped in `loop.run_in_executor`.

File: src/moonbridge/tool_handlers.py [hunk 2]
```diff
@@ -204,8 +205,16 @@ async def handle_tool(
...
                 return json_text(info)
 
+            if name == "list_models":
+                adapter = deps.get_adapter(arguments.get("adapter"))
+                provider_raw = arguments.get("provider")
+                provider = provider_raw.strip() if isinstance(provider_raw, str) else None
+                provider = provider or None
+                refresh = bool(arguments.get("refresh", False))
+                return json_text(deps.model_catalog(cwd, adapter, provider, refresh))
+
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply src/moonbridge/tool_handlers.py]: Fixed in 4497f2e — `list_models` now wrapped in `loop.run_in_executor`.

File: src/moonbridge/adapters/opencode.py [hunk 3]
```diff
@@ -91,3 +96,42 @@ def check_installed(self) -> tuple[bool, str | None]:
...
+        completed = run(
+            cmd,
+            cwd=os.path.realpath(cwd),
+            text=True,
+            capture_output=True,
+            timeout=timeout_seconds,
+            stdin=DEVNULL,
+            env={"PATH": os.environ.get("PATH", "")},
+            check=False,
+        )
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply src/moonbridge/adapters/opencode.py]: Fixed in 4497f2e — `list_models` now builds env from `self.config.safe_env_keys` (same pattern as agent execution), preserving HOME, auth keys, and config vars.

File: src/moonbridge/adapters/opencode.py [hunk 4]
```diff
@@ -91,3 +96,42 @@ def check_installed(self) -> tuple[bool, str | None]:
...
+            cmd.append("--refresh")
+
+        completed = run(
+            cmd,
+            cwd=os.path.realpath(cwd),
+            text=True,
+            capture_output=True,
+            timeout=timeout_seconds,
+            stdin=DEVNULL,
+            env={"PATH": os.environ.get("PATH", "")},
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply src/moonbridge/adapters/opencode.py]: Fixed in 4497f2e — same safe_env_keys pattern now applied to `list_models` subprocess.
